backend/src/v2/metadata/client.go

```diff
@@ -105,14 +108,22 @@ type Client struct {
 }
 
 // NewClient creates a Client given the MLMD server address and port.
-func NewClient(serverAddress, serverPort string) (*Client, error) {
+func NewClient(serverAddress, serverPort string, tlsEnabled bool) (*Client, error) {
 	opts := []grpc_retry.CallOption{
 		grpc_retry.WithMax(mlmdClientSideMaxRetries),
 		grpc_retry.WithBackoff(grpc_retry.BackoffExponentialWithJitter(300*time.Millisecond, 0.20)),
 		grpc_retry.WithCodes(codes.Aborted),
 	}
+
+	creds := insecure.NewCredentials()
+	if tlsEnabled {
+		config := &tls.Config{}
+		creds = credentials.NewTLS(config)
+	}
+
 	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", serverAddress, serverPort),
 		grpc.WithInsecure(),
+		grpc.WithTransportCredentials(creds),
```

Comment:
we can remove grpc.WithInsecure() (it's deprecated) see: 

> Deprecated: use WithTransportCredentials and insecure. NewCredentials() instead. Will be supported throughout 1.x.

since we are using: 

```go
creds := insecure.NewCredentials()
grpc.WithTransportCredentials(creds)
```
that is the equivalent of `grpc.WithInsecure()` 

We also don't want .withInsecure when we have tlsEnabled

Reply:
Done.